rust/processor/src/db/common/models/emojicoin_models/enums.rs

```diff
@@ -0,0 +1,120 @@
+use super::constants::{
+    CHAT_EVENT, GLOBAL_STATE_EVENT, LIQUIDITY_EVENT, MARKET_REGISTRATION_EVENT, MARKET_RESOURCE,
+    PERIODIC_STATE_EVENT, STATE_EVENT, SWAP_EVENT,
+};
+use serde::{Deserialize, Deserializer, Serialize};
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize, diesel_derive_enum::DbEnum,
+)]
+#[ExistingTypePath = "crate::schema::sql_types::Triggers"]
+pub enum Triggers {
+    PackagePublication,
+    MarketRegistration,
+    SwapBuy,
+    SwapSell,
+    ProvideLiquidity,
+    RemoveLiquidity,
+    Chat,
+}
+
+impl Triggers {
+    pub fn from_i16(i: i16) -> Option<Self> {
+        match i {
+            0 => Some(Self::PackagePublication),
+            1 => Some(Self::MarketRegistration),
+            2 => Some(Self::SwapBuy),
+            3 => Some(Self::SwapSell),
+            4 => Some(Self::ProvideLiquidity),
+            5 => Some(Self::RemoveLiquidity),
+            6 => Some(Self::Chat),
+            _ => None,
+        }
+    }
+}
+
+pub fn deserialize_state_trigger<'de, D>(
+    deserializer: D,
+) -> core::result::Result<Triggers, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    use serde::de::Error;
+    let trigger = <i16>::deserialize(deserializer)?;
+    match Triggers::from_i16(trigger) {
+        Some(trigger) => Ok(trigger),
+        None => Err(D::Error::custom(format!(
+            "Failed to deserialize Trigger from i16: {}",
+            trigger
+        ))),
+    }
+}
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize, diesel_derive_enum::DbEnum,
+)]
+#[ExistingTypePath = "crate::schema::sql_types::Periods"]
+pub enum Periods {
```

Comment:
How about `Period` ?

Reply:
Main reason I did this was to keep it consistent with the type in postgres, since the field would be `trigger trigger NOT NULL`, unless it doesn't matter..?